src/services/workspace-client/cheWorkspaceClient.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2018-2020 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { inject, injectable } from 'inversify';
+import { default as WorkspaceClientLib, IWorkspaceMasterApi, IRemoteAPI } from '@eclipse-che/workspace-client';
+import { EventEmitter } from 'events';
+import { WorkspaceClient } from '.';
+import { KeycloakSetupService } from '../keycloak/setup';
+
+export type WebSocketsFailedCallback = () => void;
+
+const VALIDITY_TIME = 5;
+
+/**
+ * This class manages the api connection.
+ */
+@injectable()
+export class CheWorkspaceClient extends WorkspaceClient {
+  private originLocation: string;
+  private baseUrl: string;
+  private _restApiClient: IRemoteAPI;
+  private _jsonRpcMasterApi: IWorkspaceMasterApi;
+  private _failingWebSockets: string[];
+  private webSocketEventEmitter: EventEmitter;
+  private webSocketEventName = 'websocketChanged';
+  private defaultNamespace: string;
+
+  /**
+   * Default constructor that is using resource.
+   */
+  constructor(@inject(KeycloakSetupService) keycloakSetupService: KeycloakSetupService) {
+    super(keycloakSetupService);
+    this.baseUrl = '/api';
+    this._failingWebSockets = [];
+    this.webSocketEventEmitter = new EventEmitter();
+
+    this.originLocation = new URL(window.location.href).origin;
+  }
+
+  get restApiClient(): IRemoteAPI {
+    // Lazy initialization of restApiClient
+    if (!this._restApiClient) {
+      this.updateRestApiClient();
+    }
+    return this._restApiClient;
+  }
+
+  get jsonRpcMasterApi(): IWorkspaceMasterApi {
+    // Lazy initialization of restApiClient
+    if (!this._jsonRpcMasterApi) {
+      this.updateJsonRpcMasterApi();
+    }
```

Comment:
Lazy initialization was removed in favor of eager initialization in `PreloadData.ts` (commit https://github.com/eclipse/che-dashboard/commit/cdecd9cd66998a287d61800f8690099e6598028c#)